circuitbreaker.py

```diff
@@ -62,7 +62,10 @@ def decorate(self, function):
         Applies the circuit breaker to a function
         """
         if self._name is None:
-            self._name = function.__name__
+            try:
+                self._name = function.__qualname__
+            except AttributeError:
+                self._name = function.__name__
```

Comment:
Unfortunately `__qualname__` in only available for python version >=3.3, so we need to fallback to the original case for a lower python version.

```
import functools
    
def print_name(f):
    @functools.wraps(f)
    def wrap(*args, **kwargs):
        try:
            print(f.__qualname__)
        except AttributeError:
            print(f.__name__)
        return f(*args, **kwargs)
    return wrap

class A:
    @print_name
    def b(self):
        pass

a = A()
a.b()

print("========")

@print_name
def c():
    pass

c()
```

```
$ python3 sandbox.py
A.b
========
c

$ python sandbox.py
b
========
c
```